syntax = "proto3";

service DataNode{
    rpc Store(StoreRequest) returns (StoreReply){}
    rpc Read(ReadRequest) returns (ReadReply){}
    rpc isAlive(AliveRequest) returns (AliveReply){}    
}

message StoreRequest{
    string user_name = 1;
    string file_name = 2;
    bytes file_content = 3;    
}

message StoreReply{
    string reply_msg = 1;    
}

message ReadRequest{
    string user_name = 1;
    string file_name = 2;
    string timestamp = 3;
    int32 block_size = 4;
}

message ReadReply{
    string reply_msg = 1;
    bytes reply_file = 2;    
}

message AliveRequest{
    bool ping = 1;
}

message AliveReply{
    bool health = 1;
    string reply_msg = 2;    
}
